### PR TITLE
fix panel position, add esc, enter in category setting

### PR DESCRIPTION
**Pull request by hmquannnnn**
**mouseY = event.clientY + window.screenY** will cause mouseY to be calculated incorrectly when the browser window is minimized and moved downward.

### CODE REVIEW

File: libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx [hunk 1]
```diff
@@ -1,75 +1,93 @@
...
+	useEscapeKey(() => {
+		if (hasChanged) {
+			handleReset();
+		} else {
+			onClose();
+		}
+	});
+
+	const handlePressEnter = (e: React.KeyboardEvent<HTMLInputElement>) => {
+		if (e.key === 'Enter' && !isInvalidName) {
```
Comment (nqdd) [inline libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx]: delcare enum
https://gist.github.com/cjcliffe/1185173

File: libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx [hunk 2]
```diff
@@ -1,75 +1,93 @@
...
+	}, [categoryName, category?.category_name]);
+	const dispatch = useAppDispatch();
+	const currentClan = useSelector(selectCurrentClan);
+
+	const handleChangeCategoryName = (categoryName: string) => {
+		setCategoryName(categoryName);
+
+		const regex = ValidateSpecialCharacters();
+
+		if (categoryName.length === 0 || categoryName.length === 64 || !regex.test(categoryName)) {
```
Comment (nqdd) [inline libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx]: khai bao constant cho 0 va 64
CATEGORY_MIN_LENGTH
CATEGORY_MAX_LENGTH

File: libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx (hunk 1 above)
Comment (hmquannnnn) [reply libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx]: done ạ

File: libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx (hunk 2 above)
Comment (hmquannnnn) [reply libs/components/src/lib/components/CategorySetting/OverviewSetting.tsx]: done ạ

**Comment by nqdd** [comment]
ok

**Pull request closed without merging**